R/user_error_infrastructure.R

```diff
@@ -0,0 +1,107 @@
+prepare_error_report <- function(
+  report_template = file.path("inst", "rmd", "user_errors.Rmd"),
+  error_file = file.path(html_directory, "user_errors.json"),
+  html_directory = file.path(outputs_path, portfolio_name, "report"),
+  file_name = "index.html"
+  ) {
+  if (file.exists(error_file)) {
+    errors <- jsonlite::read_json(path = error_file)
+  } else {
+    stop("user errors file cannot be found.")
+  }
+  if (!dir.exists(html_directory)) {
+    dir.create(html_directory)
+  }
+  output_file <- file.path(html_directory, file_name)
+  rmarkdown::render(
+    input = report_template,
+    output_file = output_file,
+    params = list(errors = errors)
+  )
+}
+
+log_user_errors <- function(
+  error_name,
+  suggested_action = "",
+  description = "",
+  error_file = file.path(
+    outputs_path,
+    portfolio_name,
+    "report",
+    "user_errors.json"
+    ),
+  immediate = FALSE
+  ) {
+  # check for strings
+  stopifnot(is.character(error_name))
+  stopifnot(is.character(suggested_action))
+  stopifnot(is.character(description))
+  # check that directory exists, create if not
+  if (!dir.exists(dirname(error_file))) {
+    dir.create(dirname(error_file), recursive = TRUE)
+  }
```

Comment:
Nit. The comment is redundant and, as any other comment, is at risk of eventually falling out of sync with the code.

Reply:
agreed.